src/ol/format/WFS.js

```diff
@@ -938,7 +939,17 @@ function writeProperty(node, pair, objectStack) {
         GML32.prototype.writeGeometryElement(value, pair.value, objectStack);
       }
     } else {
-      writeStringTextNode(value, pair.value);
+      if (typeof pair.value === 'string') {
+        try {
+          // Generate a CDATA section to preserve whitespaces.
+          writeCDATASection(value, pair.value);
+        } catch {
+          // If value contains the CDATA closing marker ']]>', fallback to a text node.
+          writeStringTextNode(value, pair.value);
+        }
```

Comment:
Not sure if it would work, but how about
```suggestion
      pair.value.split(']]>').forEach((text) => writeCDataSection(value, text));
```